ixmp/core.py

```diff
@@ -299,7 +306,21 @@ def add_region_synomym(self, region, mapped_to):
         mapped_to : str
             Name of the region to which the synonym should be mapped.
         """
-        self._jobj.addNodeSynonym(mapped_to, region)
+        _regions = self.regions()
+        if region not in list(_regions['region']):
+            self._jobj.addNodeSynonym(mapped_to, region)
+        else:
+            _logger_region_exists(_regions, region)
+
+
+def _logger_region_exists(_regions, r):
+    region = _regions.set_index('region').loc[r]
+    msg = 'region `{}` is already defined in the platform instance'
+    if region.mapped_to is not None:
```

Comment:
I would prefer we use explicitly indexing here (i.e., region['mapped_to']) so we all know this is a column and not an element on some custom dataframe class. Same below!

Reply:
done